mopro-ffi/src/circom/mod.rs

```diff
@@ -136,6 +158,48 @@ pub fn to_ethereum_inputs(inputs: Vec<u8>) -> Vec<String> {
     inputs
 }
 
+/// This marco will generate a Circom circuit struct named: `<circuit_name>CircomCircuit`.
+/// You can then call the `prove` and `verify` methods on this struct to interact with the circuit.
+/// Warning: Make sure that there are no other circuits with the same name that you export
+#[macro_export]
+macro_rules! mopro_circom_circuit {
+    ($name:ident) => {
+        mopro_ffi::reexports::paste! {
+
+            mod [<$name _tmp_impl_module_circom>] {
+
+                rust_witness::witness!([<$name>]);
```

Comment:
Sorry, I assumed that the `rust_witness::witness` is going to be the default implementation we will be providing. And if so, then my goal was making it as possible for the user to use the app, abstracting majority of complexity. Hence I thought that just adding `mopro_ffi::mopro_circom_circuit!(multiplier2);` is the optimal complexity level. 

I can propose a two option system: this for the simples integration, option two similar to the Halo2 approach, where the struct is required to implement a trait which in this case is going to be the `witness_func() => WtnsFn`.  Would that work?

Reply:
This will allow a user to do something like:

```rust
struct Keccak256Circuit {};

impl CircomCircuit for Keccak256Circuit {
   fn witness_func() {
       rust_witness::witness!(keccak256);
   }
}

mopro_ffi::mopro_circom_circuit!(Keccak256Circuit);
```

Reply:
IMO mopro shouldn't have a "default witness generator". mopro shouldn't care _how_ the witness is generated, only that it is. rust-witness is the _recommended_ witness generator for now, mostly for lack of better options. The witness generator should always be replaceable by the user though.

I think it's easy to confuse _shortest_ abstraction with _best_ abstraction. My problem with the two option solution here is, the easy option is _too_ abstracted, and the advanced option is _too_ complex. The easy option doesn't allow the witness function to be customized, the advanced option requires understanding of rust structs+traits.

The current solution is (arguably) simple enough that someone unfamiliar with rust can read it and have an idea of what is happening. Users need to understand the concept of a macro, and a function. A third party package can easily tell the user how to replace rust-witness with a different witness generator function, and there's enough information in the code that the user has an idea that this is possible.

I think a better solution might be a `mopro_circom` macro that accepts a `zkey_witness_map` function, instead of relying on it being in scope like the current implementation.

Reply:
I see the point, I will propose something of that sort tomorrow. What I was mostly against is the fact that we had to have a function in scope, which could have resulted in a lot of silent failures.

Reply:
I ended up proposing to use:

```rust
mopro_ffi::mopro_circom_circuit!(Multiplier2CircomCircuit, multiplier2_witness);
```
Where first parameter is the name of the struct to be defined for the circuit, the second is the WtnFc